test/19_async-migrator.ts

```diff
@@ -0,0 +1,522 @@
+			result = await tx.executeSql(`SELECT * FROM "migration status";`);
+		});
+	} catch (err) {
+		console.log(`test getMigrationStatus: ${err}`);
+	}
+	return result;
+};
+
+const initTestTableData = async function (query: string) {
+	await sbvrUtils.db.transaction(async (tx) => {
+		try {
+			await tx.executeSql(query);
+		} catch (err) {
+			console.log(`err: ${err}`);
+		}
+	});
+};
+
+const updateOneTableRow = async function (
+	tableName: string,
+	columnName: string = 'columnC',
+) {
+	await sbvrUtils.db.transaction(async (tx) => {
+		await tx.executeSql(
+			`	UPDATE "${tableName}"
+				SET "${columnName}" = 'newData'
+				WHERE id IN ( 	SELECT id FROM "${tableName}"
+								LIMIT 1);`,
+		);
+	});
+};
+
+const stopAllAsyncMigrations = async function () {
+	await sbvrUtils.db.transaction(async (tx) => {
+		await tx.executeSql(
+			`	UPDATE "migration status"
+				SET "should stop" = 1;`,
+		);
+	});
+
+	const sortedRows = (await getMigrationStatus())?.rows.sort(
+		(a, b) => a['backoffDelayMS'] - b['backoffDelayMS'],
+	);
+	await delay(sortedRows.pop()?.['backoffDelayMS'] * 2); // wait until longest backoff time has passed for all migrations to continue.
+};
+
+describe('Async Migrations', async function () {
+	this.timeout(20000);
+
+	describe('standard async migrations', function () {
+		before(async function () {
+			const testConfig = require('../config');
+
+			testConfig.models[0].migrationsPath = fixturesFolder + '01-migrations/';
+
+			// delete cached migrations
+			testConfig.models[0].migrations = {};
+			await initSupertest({ initConfig: testConfig, deleteDatabase: true });
+
+			// manually calling the init data creation sql query as the model gets
+			// executed for the first time and sync migrations are skipped
+			const initDataSql = await fs.promises.readFile(
+				testConfig.models[0].migrationsPath + '0001-init-data.sync.sql',
+				'utf8',
+			);
+
+			await initTestTableData(initDataSql);
+		});
+
+		after(async function () {
+			await stopAllAsyncMigrations();
+		});
+
+		it('should run one async migrator', async function () {
+			let result: dbModule.Result = {} as dbModule.Result;
+			// active wait until 1 row has been migrated
+			do {
+				result = await getMigrationStatus();
+			} while (result?.rows[0]?.['migrated rows'] < 1);
+
+			result = await getAllRowsEqualCols('test');
+			expect(result?.rows).to.be.not.empty;
+		});
+
+		it('should complete / catch up data in one async migrator', async function () {
+			let result: dbModule.Result = {} as dbModule.Result;
+			// active wait to check if migrations have catched up
+			await waitForAllDataMigrated('test');
+
+			result = await getMigrationStatus();
+
+			expect(result?.rows[0]?.['migrated rows']).to.equal(12);
+			// expect(result?.rows[])
+		});
+
+		it('should migrate future data change after first catch up', async function () {
+			let result: dbModule.Result = {} as dbModule.Result;
+
+			const startTime = Date.now().valueOf();
```

Comment:
`valueOf` is unnecessary:
```suggestion
			const startTime = Date.now();
```